substrate/primitives/core/src/address_uri.rs

```diff
@@ -0,0 +1,258 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Little util for parsing an address URI. Replaces regular expressions.
+
+#[derive(Debug, PartialEq)]
+/// A container for results of parsing the address uri string.
+pub struct AddressUri<'a> {
+	pub ss58: Option<&'a str>,
```

Comment:
As this could also be the `phrase`, naming it `ss58` is not totally correct.

Reply:
047eb835f1